internal/test/framework/windowsvm.go

```diff
@@ -38,70 +44,128 @@ type TestWindowsVM interface {
-// credentials will be used to interact with the VM. If no error is returned then it is guaranteed that the VM was
-// created and can be interacted with. If skipSetup is true, then configuration steps are skipped.
-func newWindowsVM(imageID, instanceType string, credentials *types.Credentials, skipSetup bool) (TestWindowsVM, error) {
-	w := &testWindowsVM{}
-	var err error
-
-	cloudProvider, err = cloudprovider.CloudProviderFactory(kubeconfig, awsCredentials, "default", artifactDir,
-		imageID, instanceType, sshKey, privateKeyPath)
+//createMachineSet() gets the generated MachineSet configuration from cloudprovider package and creates a MachineSet
+func (f *TestFramework) createMachineSet() error {
+	cloudProvider, err := providers.NewCloudProvider(sshKey)
+	if err != nil {
+		return fmt.Errorf("error instantiating cloud provider %v", err)
+	}
+	machineSet, err := cloudProvider.GenerateMachineSet(true, 1)
+	if err != nil {
+		return fmt.Errorf("error creating Windows MachineSet: %v", err)
+	}
+	log.Print("Creating Machine Sets")
+	err = f.client.Create(context.TODO(), machineSet)
 	if err != nil {
-		return nil, fmt.Errorf("error instantiating cloud provider %v", err)
+		return fmt.Errorf("unable to create MachineSet %v", err)
 	}
+	f.machineSet = machineSet
+	log.Print("Created Machine Sets")
+	return nil
+}
+
+// waitForMachines() waits until all the machines required are in Provisioned state. It returns an array of all
+// the machines created. All the machines are created concurrently.
+func (f *TestFramework) waitForMachines(vmCount int) ([]mapi.Machine, error) {
+	log.Print("Waiting for Machine Sets ")
+	windowsOSLabel := "machine.openshift.io/os-id"
+	var provisionedMachines []mapi.Machine
+	// it takes approximately 12 minutes in the CI for all the machines to appear.
+	timeOut := 12 * time.Minute
+	startTime := time.Now()
+	for i := 0; time.Since(startTime) <= timeOut; i++ {
+		allMachines := &mapi.MachineList{}
 
-	if credentials == nil {
-		windowsVM, err := cloudProvider.CreateWindowsVM()
+		err := f.client.List(context.TODO(), allMachines, client.InNamespace("openshift-machine-api"), client.HasLabels{windowsOSLabel})
 		if err != nil {
-			return nil, fmt.Errorf("error creating Windows VM: %v", err)
+			return nil, fmt.Errorf("failed to list machines: %v", err)
 		}
-		// TypeAssert to the WindowsVM struct we want
-		winVM, ok := windowsVM.(*types.Windows)
-		if !ok {
-			return nil, fmt.Errorf("error asserting Windows VM: %v", err)
-		}
-		w.Windows = winVM
-	} else {
-		//TODO: Add username as well, as it will change depending on cloud provider
-		if credentials.GetIPAddress() == "" || credentials.GetPassword() == "" {
-			return nil, fmt.Errorf("password or IP address not specified in credentials")
+		provisionedMachines = []mapi.Machine{}
+
+		phaseProvisioned := "Provisioned"
+
+		for _, machine := range allMachines.Items {
+			instanceStatus := machine.Status
```

Comment:
Can we combine these conditions to one -
`if instanceStatus.Phase != nil && *instanceStatus.Phase == phaseProvisioned{
    append()
}`